### PR TITLE
Use latest codeclimate config and update dependencies

### PR DESCRIPTION
**Pull request by brwnrclse**
Fix packages that were flagged from `npm audit`
Also update the codeclimate config to the latest spec

### CODE REVIEW

File: .codeclimate.yml [hunk 1]
```diff
@@ -1,10 +1,9 @@
 ---
-engines:
+version: "2"
```
Comment (brwnrclse) [inline .codeclimate.yml]: https://docs.codeclimate.com/docs/advanced-configuration

File: docs/utils_setConfig.js.html [hunk 2]
```diff
@@ -125,7 +124,7 @@ <h1 class="page-title">Source: utils/setConfig.js</h1>
 	// if 2, we pass in a path to the config
 	// this only occurs if using stylint via the command line
 	else if ( configpath ) {
-		customPath = pathIsAbsolute( configpath ) ? configpath : cwd + '/' + configpath
+		customPath = path.isAbsolute( configpath ) ? configpath : cwd + '/' + configpath
```
Comment (brwnrclse) [inline docs/utils_setConfig.js.html]: https://nodejs.org/api/path.html

File: src/utils/setConfig.js [hunk 3]
```diff
@@ -3,7 +3,6 @@
 var fs = require( 'fs' )
 var path = require( 'path' )
 var userHome = require( 'user-home' )
-var pathIsAbsolute = require( 'path-is-absolute' )
```
Comment (brwnrclse) [inline src/utils/setConfig.js]: https://nodejs.org/api/path.html

File: package.json [hunk 4]
```diff
@@ -36,32 +36,30 @@
...
   },
   "scripts": {
-    "cover": "istanbul cover node_modules/.bin/_mocha -- -u exports -R spec test/test.js",
     "docs": "jsdoc src/ --recurse --destination docs/",
     "lint": "eslint .",
     "precover": "npm run lint",
     "pretest": "npm run lint",
     "start": "node --harmony index.js",
-    "test": "mocha"
+    "test": "nyc mocha"
```
Comment (brwnrclse) [inline package.json]: https://istanbul.js.org/

**Comment by brwnrclse** [comment]
@SimenB mind taking a look at this when you get a chance?

File: package.json (hunk 4 above)
Comment (SimenB) [reply package.json]: keep `cover` but run `nyc` there?

File: .travis.yml [hunk 5]
```diff
@@ -1,10 +1,7 @@
 language: node_js
 node_js:
   - node
```
Comment (SimenB) [inline .travis.yml]: add `10` and `8` here as well?

File: package.json (hunk 4 above)
Comment (brwnrclse) [reply package.json]: SGTM

File: .travis.yml (hunk 5 above)
Comment (brwnrclse) [reply .travis.yml]: SGTM

File: .travis.yml [hunk 6]
```diff
@@ -1,10 +1,9 @@
...
-  - 4
-  - 0.12
-  - "0.10"
-script: npm run cover
+  - 8
+  - 12
```
Comment (SimenB) [inline .travis.yml]:
```suggestion
  - 10
```

`12` is covered by `node` 🙂

File: package.json [hunk 7]
```diff
@@ -36,27 +36,26 @@
...
     "mocha": "2.5.3",
+    "nyc": "^14.1.1",
     "sinon": "1.17.4",
     "touch": "1.0.0"
   },
   "dependencies": {
     "async": "1.5.2",
     "chalk": "1.1.3",
-    "chokidar": "1.5.2",
+    "chokidar": "^3.0.1",
```
Comment (SimenB) [inline package.json]:
```suggestion
    "chokidar": "3.0.1",
```

for consistency

File: .travis.yml (hunk 6 above)
Comment (brwnrclse) [reply .travis.yml]: Oh ha my bad

**Pull request merged**